parsel/selector.py

```diff
@@ -50,10 +50,14 @@ def _st(st: Optional[str]) -> str:
         raise ValueError("Invalid type: %s" % st)
 
 
-def create_root_node(text, parser_cls, base_url=None):
+def create_root_node(text, parser_cls, base_url=None, encoding="utf8"):
     """Create root node for text using given parser class."""
-    body = text.strip().replace("\x00", "").encode("utf8") or b"<html/>"
-    parser = parser_cls(recover=True, encoding="utf8")
+    if isinstance(text, bytes):
+        body = text.replace(b"\x00", b"").strip()
+    else:
+        body = text.replace("\x00", "").strip().encode(encoding)
+    body = body or b"<html/>"
```

Comment:
Technically, the old code supported passing text=None but this one doesn't (no idea if we should care)

Reply:
`text.strip()` from the old code would have failed right away in that scenario.